Assets/Scripts/Game/EnablePlayerTorch.cs

```diff
@@ -24,35 +25,76 @@ public class EnablePlayerTorch : MonoBehaviour
 
         DaggerfallUnity dfUnity;
         PlayerEnterExit playerEnterExit;
+        PlayerEntity playerEntity;
         Light torchLight;
         float torchIntensity;
+        float intensityMod = 1f;
+        float lastTickTime;
+        float tickTimeInterval = 10f;
 
         void Start()
         {
             dfUnity = DaggerfallUnity.Instance;
             playerEnterExit = GetComponent<PlayerEnterExit>();
+            playerEntity = GameManager.Instance.PlayerEntity;
 
             if (PlayerTorch)
             {
                 torchLight = PlayerTorch.GetComponent<Light>();
                 if (torchLight)
+                {
                     torchIntensity = torchLight.intensity;
+                    if (DaggerfallUnity.Settings.PlayerTorchFromItems)
+                    {
+                        torchLight.shadows = LightShadows.Soft;
+                        torchLight.range = 12;
+                    }
+                }
             }
         }
 
         void Update()
         {
-            if (!dfUnity.IsReady || !playerEnterExit || !PlayerTorch)
+            if (!dfUnity.IsReady || !playerEnterExit || !PlayerTorch || playerEntity == null)
                 return;
 
             bool enableTorch = false;
-            if (!playerEnterExit.IsPlayerInside && dfUnity.WorldTime.Now.IsCityLightsOn)
-                enableTorch = true;
-            if (playerEnterExit.IsPlayerInsideDungeon && !playerEnterExit.IsPlayerInsideDungeonCastle)
-                enableTorch = true;
-
+            if (DaggerfallUnity.Settings.PlayerTorchFromItems && !GameManager.IsGamePaused)
+            {
+                DaggerfallUnityItem lightSource = playerEntity.LightSource;
+                if (lightSource != null)
+                {
+                    enableTorch = true;
+                    // Consume durability / fuel
+                    if (Time.realtimeSinceStartup > lastTickTime + tickTimeInterval)
+                    {
+                        lastTickTime = Time.realtimeSinceStartup;
+                        if (lightSource.currentCondition > 0)
+                            lightSource.currentCondition--;
+                        else
+                        {
+                            DaggerfallUI.MessageBox("Your %it flickers and dies.", false, lightSource);
```

Comment:
It would be great to move text like this to a text table please. I'm trying to use them as much as possible moving forwards.

Reply:
Sorry, must have missed that one - thought I had done 'em all.